### DISCUSSION TITLE
Unable to set path in access phase using kong.service.request.set_path (Custom Plugin)

### DISCUSSION THREAD

**Discussion by sivasubramani300890** [Kong/kong#9803] [Help] [answered]
Hi Team,

Trying to set path in existing hostname in access phase, but its not happening

Existing : http://www.abc.com/common
New : http://www.abc.com/common/devices

Using kong.service.request.set_path('/devices') to set the path

**Comment by StarlightIbuki** [comment, marked as answer]
`kong.service.request` is supposed to be used to manipulate the request to upstream (change `upstream_uri` in this case), while I guess you are fetching the path with `kong.request.get_path`. We do not support request modification which is visible to later PDK calls. (It may confuse people as later phases would use them to route the request).

If you want to test if the call is in effect, please verify from the upstream server.

If you really want to change the request path, you may call `ngx.request.set_path`. Ref: https://github.com/openresty/lua-nginx-module#ngxreqset_uri

----
#### Before Updated
Thanks for reporting. Could you explain how do you find it's not happening?

**Reply by sivasubramani300890** [reply]
Using **python** to develop third party plugin

**Reply by StarlightIbuki** [reply]
Python (and Go/JS) PDK does not yet support this. Actually, we do not encourage direct modification to the request as it may interfere with other plugins. If you really want to do this, please write a custom Lua plugin.

**Reply by sivasubramani300890** [reply]
If i want just to remove any character,Can we do the same by using Request Transformer Plugin (config.replace.uri) ?

**Reply by StarlightIbuki** [reply]
> uri

The Request Transformer Plugin does the same as `kong.service.request.set_path`.

**Comment by sivasubramani300890** [comment]
Using kong.service.request.set_path('/devices') updating path, if i display the path it shows existing only not updated

Using kong.response.exit to display path

**Comment by sivasubramani300890** [comment]
The problem is when i am trying to setup new path instead of existing path in upstream its not considering

for example :

Existing path : [www.abc.com/v1/data](http://www.abc.com/v1/data)
New Path : [www.abc.com/v2/data?key=value](http://www.abc.com/v2/data?key=value)

Tried using kong.service.request.set_path("NewPath"),but didn't worked
